### PR TITLE
if judgement always true in e2e

### PR DESCRIPTION
**Pull request by jianglingxia**
**What this PR does / why we need it**:
the if judge always true so remove the var and fix defer func
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jsafrane** [comment]
/assign @eparis @liggitt 
for approval

**Comment by jianglingxia** [comment]
@liggitt for approval again because of need rebase,thanks

image

File: test/e2e/common/volumes.go [hunk 1]
```diff
@@ -137,11 +127,9 @@ var _ = Describe("[sig-storage] GCP Volumes", func() {
 			config, _, _ := framework.NewGlusterfsServer(c, namespace.Name)
 			name := config.Prefix + "-server"
 			defer func() {
-				if clean {
-					framework.VolumeTestCleanup(f, config)
-					err := c.CoreV1().Endpoints(namespace.Name).Delete(name, nil)
-					Expect(err).NotTo(HaveOccurred(), "defer: Gluster delete endpoints failed")
-				}
+				framework.VolumeTestCleanup(f, config)
+				err := c.Core().Endpoints(namespace.Name).Delete(name, nil)
```
Comment (liggitt) [inline test/e2e/common/volumes.go]: CoreV1

Comment (jianglingxia) [reply test/e2e/common/volumes.go]: thanks,I amend the error and wait for your review again ,for your lgtm!thanks very much!

**Pull request merged**